File: python/excepciones.py
class Error(Exception):
    pass


class NumArgsError(Exception):
    def __init__(self, num):
        self.num = num
        self.message = (
            "Error en el número de argumentos. Deben ser "
            + str(num)
            + " valores separados por un espacio."
        )
        super().__init__(self.message)

    def __str__(self) -> str:
        return self.message


class CoordenateError(Exception):
    def __init__(
        self, message="Error en el input de coordenadas. Tienen que ser valores enteros"
    ):
        self.message = message
        super().__init__(self.message)

    def __str__(self) -> str:
        return self.message


class Max50ValueError(Exception):
    def __init__(
        self, message="Error, el valor máximo de cualquier coordenada debe ser 50."
    ):
        self.message = message
        super().__init__(self.message)

    def __str__(self) -> str:
        return self.message


class OrientationError(Exception):
    def __init__(
        self,
        message="La orientacion introducida es incorrecta. Debe ser un caracter en mayúscula que represente un punto "
        "cardinal",
    ):
        self.message = message
        super().__init__(message)

    def __str__(self) -> str:
        return self.message


class InstruccError(Exception):
    def __init__(
        self,
        message="Las instrucciones introducidas son incorrectas. Solo se aceptan letras que referencien a instrucciones implementadas.",
    ):
        self.message = message
        super().__init__(message)

    def __str__(self) -> str:
        return self.message

File: node/src/logica/excepciones.js
class NumArgsError extends Error {

    constructor(num) {
        let mensaje = "Error en el número de argumentos. Deben ser " + num.toString() +
            " valores separados por un espacio."
        super(mensaje);
        this.name = NumArgsError
        this.message = "Error en el número de argumentos. Deben ser " + num.toString() +
            " valores separados por un espacio."
    }

    toString() {
        return this.message
    }
}

class CoordenateError extends Error {

    constructor() {
        super();
        this.message = "Error en el input de coordenadas. Tienen que ser valores enteros"
    }

    toString() {
        return this.message
    }
}

class Max50ValueError extends Error {
    constructor() {
        super();
        this.message = "Error, el valor máximo de cualquier coordenada debe ser 50."
    }

    toString() {
        return this.message
    }
}

class OrientationError extends Error {
    constructor() {
        super();
        this.message = "La orientacion introducida es incorrecta. " +
            "Debe ser un caracter en mayúscula que represente un punto cardinal"
    }

    toString() {
        return this.message
    }
}

class InstruccError extends Error {
    constructor() {
        super();
        this.message = "Las instrucciones introducidas son incorrectas. " +
            "Solo se aceptan letras que referencien a instrucciones implementadas y menos de 100 caracteres."
    }

    toString() {
        return this.message
    }
}


module.exports = {
    NumArgsError: NumArgsError,
    CoordenateError: CoordenateError,
    Max50ValueError: Max50ValueError,
    OrientationError: OrientationError,
    InstruccError: InstruccError

}